core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2712,6 +2714,48 @@ public static String jsonType(Object o) {
     }
   }
 
+  public static Integer jsonDepth(Object o) {
+    final Integer result;
+    try {
+      if (o == null) {
+        result = null;
+      } else {
+        result = getJsonDepth(o);
+      }
+      return result;
+    } catch (Exception ex) {
+      throw RESOURCE.unknownObjectOfJsonDepth(o.toString()).ex();
+    }
+  }
+
+  public static Integer getJsonDepth(Object o) {
```

Comment:
It looks like this function should be private as it's only used by `jsonDepth` above.